### PR TITLE
trunk styles: add scripts to gather style files to push them into a separated repo

### PR DESCRIPTION
**Pull request by daniwe4**
Description by @klees:
-------------------------------
Dear colleagues,

when discussing the implications of #5102 for skinning (thx @Amstutz, @catenglaender, @ezenzen), we found that it would be a huge improvement for people with advanced skinning requirements to have all code for delos in one repository. This will:

* allow to monitor delos for changes
* apply patches to delos
* provide a starting point for individual skins

Still we did not consider it a good move to just move delos from the ILIAS repo to its own repo, as the templates and styles are intimately tied to the UI components (and legacy code) that requires them. Hence, changes always would need to be coordinated between two repositories if we had delos and the other ILIAS code in separate repos.

Hence an automatism: On every commit, this collects all files relevant for a skin and pushes them to a separate repo.

At some point in the near future someone (@catenglaender maybe?) will add a description on how to use the repo, this only implements the automatism. Feel free to ask questions, voice concerns, etc. until then, as always.

Thanks!

Remark by @daniwe4:
-------------------------------
Please be aware, that the REPO variable in CI/Style-To-Repo/deploy.sh has to be set to the new style repo. The actual value of this variable is only a test dummy.

### CODE REVIEW

**Comment by klees** [comment]
Hi @Amstutz,

I assign this to you as style-coordinator. I did not look through code in detail, so a review from me will follow...

Best regards!

File: .github/workflows/style-to-repo_trunk.yml
Comment (klees) [inline .github/workflows/style-to-repo_trunk.yml]: I think we should target every commit, since there might be changes in style code that are not processed via PRs.

File: CI/Style-To-Repo/build-and-deploy.sh
Comment (klees) [inline CI/Style-To-Repo/build-and-deploy.sh]: Please use our standard copyright in scripts as well. Thanks!

File: CI/Style-To-Repo/build-and-deploy.sh
Comment (klees) [inline CI/Style-To-Repo/build-and-deploy.sh:36]: Please use functions for the contents of these files as well and then call functions here. Remember that function names are a super cheap way of documenting code...

File: CI/Style-To-Repo/deploy.sh
Comment (klees) [inline CI/Style-To-Repo/deploy.sh]: Please reference the commit from the original repo: "Style changes from $HASH", then "Original message: ..." and link to github in the extended commit message.

File: CI/Style-To-Repo/deploy.sh
Comment (klees) [inline CI/Style-To-Repo/deploy.sh]: Please use a branch (as base and as target) that has the very same name as the branch that the commit in the ILIAS repo was made on. If that branch does not exist on remote, it should be created.

**Review by klees — CHANGES_REQUESTED** [review]
Hi @daniwe4,

I marked some change requests inline, feel free to contact me for further explanations.

Thanks!

**Review by Amstutz — APPROVED** [review]
Hi @klees and @daniwe4 

I approve the changes on a conceptual level. I am not deep enough into our deployment and sh scripts to perform good review of the implementation details. 

@klees : This looks good to me, feel free to merge to trunk as soon as all your feedback has been taken care of.

Thx to everybody involved!

**Comment by chfsx** [comment]
As a technical board, we regularly check whether pull requests have been open for a long time.
Are there any updates on this? Please note that you can also close this request if you are not able or if you do not have the resources to look into it in detail.

At this point we would like to refer to the rules we all agreed on: https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/contributing.md#rules-for-maintainers-assigned-to-prs.

With kind regards!

**Comment by klees** [comment]
Hi everyone,

I think this is good to be discussed on the JF, hence I added the label. I would kindly ask @tbongers-cat to test this thoroughly after his vacation.

Best wishes!

**Comment by matthiaskunkel** [comment]
JourFixe, 12 JUN 2023: We highly appreciate this suggestion and accept the PR for trunk. We will add a new Mantis category '¥ Automation' to collect issues related to this script.

File: CI/Style-To-Repo/deploy.sh
Comment (tbongers-cat) [inline CI/Style-To-Repo/deploy.sh]: `git clone` needs to use **${REPO_TOKEN}** as well (instead of **${STYLE_REPO}**), as we don't have SSH available for cloning.

File: CI/Style-To-Repo/deploy.sh
Comment (tbongers-cat) [inline CI/Style-To-Repo/deploy.sh]: This should be `"${BRANCH_EXISTS}" == "0"` afaik

File: CI/Style-To-Repo/deploy.sh
Comment (tbongers-cat) [inline CI/Style-To-Repo/deploy.sh]: Text proposal: "Detect**ed** new branch XYZ, **which** will ..."

File: CI/Style-To-Repo/deploy.sh
Comment (tbongers-cat) [inline CI/Style-To-Repo/deploy.sh]: Text proposal: "Detect**ed** changes on style files, **which** will ..."

File: CI/Style-To-Repo/build-and-deploy.sh
Comment (tbongers-cat) [inline CI/Style-To-Repo/build-and-deploy.sh]: "Build" instead of "Building" to match the notation below.

**Review by tbongers-cat — CHANGES_REQUESTED** [review]
Works as intended in my tests, but needs the **change in `deploy.sh`** as detailed in the comments. A **code rebase** should be done before merging this.

With these changes, I successfully initiated the automatic cloning of all gathered style files to a new branch on the style repo as well as the cloning of any follow-up changes to style files in the original repo. Changes to not style-related files are ignored as expected.

The other comments are just proposals for textual changes =)

File: CI/Style-To-Repo/deploy.sh
Comment (daniwe4) [reply CI/Style-To-Repo/deploy.sh]: DONE!

File: CI/Style-To-Repo/deploy.sh
Comment (daniwe4) [reply CI/Style-To-Repo/deploy.sh]: Done!

File: CI/Style-To-Repo/deploy.sh
Comment (daniwe4) [reply CI/Style-To-Repo/deploy.sh]: Done!

File: CI/Style-To-Repo/build-and-deploy.sh
Comment (daniwe4) [reply CI/Style-To-Repo/build-and-deploy.sh]: Done!

**Review by tbongers-cat — APPROVED** [review]
lgtm

**Pull request merged**
